pandas/core/dtypes/cast.py

```diff
@@ -1162,3 +1162,23 @@ def construct_1d_arraylike_from_scalar(value, length, dtype):
         subarr.fill(value)
 
     return subarr
+
+
+def construct_1d_object_array_from_listlike(values):
+    """
+    Transform any list-like object in a 1-dimensional numpy array of object
+    dtype.
+
+    Parameters
+    ----------
+    values : any iterable which has a len()
+
+    Returns
+    -------
+    1-dimensional numpy array of dtype object
+    """
```

Comment:
would not object to an
``assert is_iterable(values)`` with a nice error message

Reply:
The thing is: I can't think of any code path which could be hitting it. Scalar input to a ``Series()`` is (considered valid and) recasted to a 1-d before calling this. Similarly, an operation such as ``Series([1,2]) + 3`` transforms 3 before hitting this. So I don't know what the error message could actually say.

Reply:
not what i am asking

this is a completely internal
routine
it should fail with invalid input

Reply:
> it should fail with invalid input

Sure it does, ``TypeError: object of type 'int' has no len()``. Which is pretty clear, considering the docstring, and precisely in light of the fact that this is an internal routine. That said, feel free to suggest an error message which is worth the cost of the additional ``assert``.

Reply:
add a raises section to the doc-string